app/scripts/modules/kubernetes/src/rawResource/component/group/RawResourceDetails.tsx

```diff
@@ -0,0 +1,232 @@
+      loading: true,
+      manifest: null,
+    };
+  }
+  private destroy$ = new Subject();
+  private props$ = new Subject<IRawResourceDetailsProps>();
+  private unsubscribeManifest?: () => void;
+
+  public componentDidMount() {
+    this.setState({
+      account: this.props.$stateParams.account,
+      application: this.props.$stateParams.application,
+      name: this.props.$stateParams.name,
+      region: this.props.$stateParams.region,
+    });
+    const params = {
+      account: this.props.$stateParams.account,
+      location: this.props.$stateParams.region == '' ? '_' : this.props.$stateParams.region,
+      name: this.props.$stateParams.name,
+    };
+    this.unsubscribeManifest = KubernetesManifestService.subscribe(this.props.app, params, (manifest) => {
+      if (this.unsubscribeManifest != null) {
+        this.setState({ manifest, loading: false });
+      }
+    });
+  }
+
+  public componentWillReceiveProps(nextProps: IRawResourceDetailsProps) {
+    this.props$.next(nextProps);
+  }
+
+  public componentWillUnmount() {
+    this.destroy$.next();
+    if (this.unsubscribeManifest) {
+      this.unsubscribeManifest();
+      this.unsubscribeManifest = null;
+    }
+  }
+
+  private editRawResource() {
+    const { app } = this.props;
+    const { account, application, manifest } = this.state;
+
+    KubernetesManifestCommandBuilder.buildNewManifestCommand(
+      app,
+      manifest.manifest,
+      { app: application },
+      account,
+    ).then((builtCommand) => {
+      ManifestWizard.show({ title: 'Edit Manifest', application: app, command: builtCommand });
+    });
+  }
+
+  private deletedRawResource = () => {
+    const { app } = this.props;
+    const { account, manifest } = this.state;
+    const name = get(manifest, ['manifest', 'metadata', 'name'], null);
+    const kind = get(manifest, ['manifest', 'kind']);
+    const taskMonitor = {
+      application: app,
+      title: 'Deleting ' + name,
+    };
+
+    const submitMethod = () => {
+      ReactGA.event({ category: 'RawResource', action: 'Delete clicked' });
+      const command = {
+        manifestName: this.state.name,
+        location: this.state.region,
+        account: this.state.account,
+        reason: null as string,
+        cloudProvider: 'kubernetes',
+        options: {
+          cascading: false,
+        },
+      };
+      return ManifestWriter.deleteManifest(command, this.props.app);
+    };
+
+    ConfirmationModalService.confirm({
+      header: 'Are you sure you want to delete the ' + kind + ' ' + name + '?',
+      buttonText: 'Delete',
+      account: account,
+      taskMonitorConfig: taskMonitor,
+      submitMethod,
+    });
+  };
+
+  private handleEditClick = (): void => {
+    ReactGA.event({ category: 'RawResource', action: 'Edit clicked' });
+    this.editRawResource();
+  };
+
+  public render() {
+    const { account, region, manifest } = this.state;
+    const kind = get(manifest, ['manifest', 'kind'], null);
```

Comment:
If desired, type safety can be retained by writing these as:

```suggestion
    const kind = manifest?.manifest?.kind;
```